src/foam/nanos/boot/Boot.java

```diff
@@ -107,6 +117,34 @@ public Object create(X x) {
       }
     });
 
+    // Load hierarchically nspecs eg. "foo.bar.serviceName":
+    // 1. Create "foo" proxy sub context from the root context
+    // 2. Create "foo.bar" proxy sub context from "foo" sub context
+    // 3. Create "serviceName" object in "foo.bar" sub context
+    for ( var sp : hnSpecs ) {
+      var x = root_;
+      var pos = sp.getName().indexOf(".");
+      while ( pos > 0 ) {
```

Comment:
Easier to split().

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/5037/commits/7089465493f5f035e1ad72114e3f32d538240148.